source/Ship.h

```diff
@@ -183,7 +182,7 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// Get a random hail message, or set the object used to generate them. If no
 	// object is given the government's default will be used.
 	void SetHail(const Phrase &phrase);
-	std::string GetHail(const PlayerInfo &player) const;
+	std::string GetHail(std::map<std::string, std::string> subs) const;
```

Comment:
Make this take an rvalue
```suggestion
	std::string GetHail(std::map<std::string, std::string> &&subs) const;
```
That way we prevent a copy from being unknowingly made if e.g. this method is called with a const map